File: Behavioral/Lab21/Typescript/src/Strategies/add_apartment_strategy.ts
import { Agency } from "../Entities/agency";
import { Apartment } from "../Entities/apartment";
import { AbstractStrategy } from "./abstract_strategy";

export class AddApartmentStrategy extends AbstractStrategy {
    constructor(...params: string[]) {
        super(...params);
    }
    public execute(): string {
        const agency = Agency.GetInstance();
        if (this.Params.length === 0) {
            agency.AddEstate(new Apartment());
            return "Added empty new Apartment";
        } else if (this.Params.length === 6) {
            agency.AddEstate(new Apartment(
                this.Params[0],
                this.Params[1],
                +this.Params[2],
                +this.Params[3],
                this.Params[4],
                +this.Params[5]
            ));
            return "Added new Apartment with args: " + this.Params.join(", ");
        }
        return "Failed to add new Apartment with args: " + this.Params.join(", ");
    }
}

File: Behavioral/Lab21/C#/Strategies/AddApartmentStrategy.cs
using System;
using System.Collections.Generic;
using System.Text;
using C_.Entities;

namespace C_.TemplateMethods
{
    public class AddApartmentStrategy : AbstractStrategy
    {
        public AddApartmentStrategy(params string[] args) : base(args)
        {
        }

        public override string Execute()
        {
            var agency = Agency.GetInstance();
            if(_params.Length == 0){
                agency.AddEstate(new Apartment());
                return "Added empty new Apartment";
            }
            else if(_params.Length == 6){
                agency.AddEstate(new Apartment(_params[0], _params[1], Convert.ToDouble(_params[2]), Convert.ToDouble(_params[3]), _params[4], Convert.ToInt16(_params[5])));
                return "Added new Apartment with args: " + string.Join(", ", _params);
            }
            return "Failed to add new Apartment with args: " + string.Join(", ", _params);
        }
    }
}

